### PR TITLE
FIX : read fids in eeglab

### LINKED ISSUE #10474: read_raw_eeglab does not read nasion, lpa and rpa values from .set file
#### Describe the bug

`read_raw_eeglab` does not read properly the nasion, lpa and rpa coordinates of a `.set` file.

#### Steps to reproduce

This can be reproduced on the sample data `eeglab_data.set` file that comes with EEGLAB.

```
>>> import mne
>>> import os
>>> from mne_bids import write_raw_bids, BIDSPath
>>> import eeglabio
>>> file = "/Users/cmadjar/Documents/MATLAB/eeglab2022.0/sample_data/eeglab_data.set"
>>> raw = mne.io.read_raw_eeglab(input_fname=file, preload=False, verbose=True)
Reading /Users/cmadjar/Documents/MATLAB/eeglab2022.0/sample_data/eeglab_data.fdt
>>> 
>>> raw.get_montage()
<DigMontage | 0 extras (headshape), 0 HPIs, 0 fiducials, 32 channels>

>>> raw.get_montage().get_positions()
.... 'coord_frame': 'head', 'nasion': None, 'lpa': None, 'rpa': None, 'hsp': None, 'hpi': None
```

Screen Shot 2022-03-30 at 1 11 21 PM

However, when looking in the Matlab structure, the coordinates for nasion, lpa and rpa are indeed present in the file. So for some reason, this gets lost when loading the `.set` file with `read_raw_eeglab`

#### Additional information

Reproduced on several versions of mne and different OS.

Latest information:
```
>>> mne.sys_info()
Platform:         macOS-11.3-x86_64-i386-64bit
Python:           3.9.12 (main, Mar 26 2022, 15:52:10)  [Clang 13.0.0 (clang-1300.0.29.30)]
Executable:       /Users/cmadjar/Development/python_venv/python_intel_3.9.4/eeg2bids_py3.9.12/bin/python
CPU:              i386: 8 cores
Memory:           Unavailable (requires "psutil" package)
mne:              1.0.0
numpy:            1.21.5 {blas=openblas, lapack=openblas}
scipy:            1.8.0
matplotlib:       3.5.1 {backend=MacOSX}

sklearn:          1.0.2
numba:            0.55.1
nibabel:          Not found
nilearn:          Not found
dipy:             Not found
cupy:             Not found
pandas:           1.4.1
pyvista:          Not found
pyvistaqt:        Not found
ipyvtklink:       Not found
vtk:              Not found
PyQt5:            Not found
ipympl:           Not found
pooch:            v1.6.0

mne_bids:         0.10
mne_nirs:         Not found
mne_features:     0.2
mne_qt_browser:   Not found
mne_connectivity: Not found
```

### PR DESCRIPTION
**Pull request by agramfort**
closes #10474

@cmadjar this should fix your problem.

### CODE REVIEW

File: mne/io/eeglab/eeglab.py [hunk 1]
```diff
@@ -144,10 +144,25 @@ def _get_montage_information(eeg, get_pos):
              '\n'.join([f'{key}: {sorted(unknown_types[key])}'
                         for key in sorted(unknown_types)]))
 
+    lpa, rpa, nasion = None, None, None
+    if (hasattr(eeg, "nodatchans") and
+            "nodatchans" in eeg.chaninfo and
+            len(eeg.chaninfo['nodatchans'])):
```
Comment (hoechenberger) [inline mne/io/eeglab/eeglab.py:150]:
```suggestion
           eeg.chaninfo['nodatchans']):
```

**Comment by hoechenberger** [comment]
SphinxWindows failure is unrelated

**Review by hoechenberger — APPROVED** [review]
nitpick, otherwise LGTM!

File: mne/io/eeglab/tests/test_eeglab.py [hunk 2]
```diff
@@ -454,3 +455,14 @@ def test_get_montage_info_with_ch_type():
...
+
+@testing.requires_testing_data
+def test_fidsposition_information():
+    """Test reading file with 3 fiducial locations."""
+    raw = read_raw_eeglab(raw_fname_chanloc_fids)
+    montage = raw.get_montage()
+    pos = montage.get_positions()
+    assert pos['nasion'] is not None
+    assert pos['lpa'] is not None
+    assert pos['rpa'] is not None
```
Comment (hoechenberger) [inline mne/io/eeglab/tests/test_eeglab.py:468]: Should we maybe check whether the arrays have the correct dimension too?

File: mne/io/eeglab/eeglab.py (hunk 1 above)
Comment (agramfort) [reply mne/io/eeglab/eeglab.py:150]: I find the len more explicit

**Comment by hoechenberger** [comment]
Is the coreg failure related? (I don't think so?)
If not, feel free to merge!

**Comment by agramfort** [comment]
@larsoner no idea about these CIs failures. @larsoner merge if happy

**Comment by larsoner** [comment]
Thanks @agramfort

**Comment by cmadjar** [comment]
You guys are so fast. I only got the chance to test the PR today. I confirm that reading the fids works now for our SET file. 👍 

However, I do not see them being written again in the SET file when saving the data again and opening them in EEGLAB. If I reopen the new file with read_raw_eeglab, I can confirm that the fids are not there either. @agramfort should I open a different issue for that?

**Comment by larsoner** [comment]
> should I open a different issue for that?

Sure, that's an exporting rather than reading bug. Or you can reopen your old issue an re-title it to discuss round-trip failing. (Now the read works, but the write doesn't.)

**Comment by cmadjar** [comment]
See https://github.com/mne-tools/mne-python/issues/10529 for the new issue. I linked the old issue in the description as well.

**Comment by agramfort** [comment]
we don't have a strong support for exporting. Why do you need to write it
back to eeglab format?

**Comment by hoechenberger** [comment]
Couldn't EEGLAB's BIDS export be an option for you, then?

**Comment by cmadjar** [comment]
Not really unfortunately as it requires a Matlab license...

**Comment by hoechenberger** [comment]
And if you try our BrainVision export? Would that maybe work? Maybe it works better :)

**Comment by cmadjar** [comment]
our collaborator wants SET files... We have no other choice than to have the mne export functionality to work with SET files. Sorry

**Comment by agramfort** [comment]
@cmadjar you need to report this issue here https://github.com/jackz314/eeglabio/issues

we use this code to export

**Comment by cmadjar** [comment]
ok. Done!

**Pull request merged**